examples/c/example.c

```diff
@@ -587,6 +674,208 @@ AranyaError run_aqc_example(Team *t) {
+        EXPECT("error getting role ID", err);
+        const char *role_str;
+        err = aranya_get_role_name(&roles[i], &role_str);
+        printf("revoking perms for role: %s\r\n", role_str);
+
+        size_t perms_len  = 0;
+        AranyaPerm *perms = NULL;
+        err               = query_role_perms(t, &role_id, &perms, &perms_len);
+        printf("perms_len: %zu\r\n", perms_len);
+        EXPECT("error querying role permissions", err);
+        for (size_t j = 0; j < perms_len; j++) {
+            bool eq = false;
+            aranya_cmp_role_ids(&owner_role_id, &role_id, &eq);
+            if (!eq) {
+                // TODO: capacity overflow error
+                err = aranya_revoke_role_perm(&t->clients.owner.client, &t->id,
+                                              &role_id, &perms[j]);
+                EXPECT("error revoking role perm", err);
+                const char *perm_str;
+                err = aranya_get_perm_name(&perms[j], &perm_str);
+                EXPECT("error getting perm name", err);
+                printf("revoked role perm: %s\r\n", perm_str);
+            }
+        }
+        free(perms);
+    }
+    free(roles);
+
+    // Delete roles
+    // TODO: obtain team roles from query.
+    for (int i = 0; i < NUM_ROLES; i++) {
+        err = aranya_delete_role(&t->clients.owner.client, &t->id,
+                                 &t->roles_arr[i]);
+        EXPECT("unable to delete role", err);
+    }
+
+    return err;
+}
+
+// Query devices on team. Returned `devices` ptr must be freed.
+AranyaError query_devices_on_team(Team *t, AranyaDeviceId **devices,
+                                  size_t *devices_len) {
+    AranyaError err;
+
+    *devices_len = BUF_LEN;
+    *devices     = malloc(*devices_len * sizeof(AranyaDeviceId));
+    err = aranya_query_devices_on_team(&t->clients.operator.client, &t->id,
+                                       *devices, devices_len);
+    EXPECT("error querying devices on team", err);
+    if (devices == NULL) {
+        return ARANYA_ERROR_BUG;
+    }
+
+    return err;
+}
+
+// Query roles on team. Returned `roles` ptr must be freed.
+AranyaError query_roles_on_team(Team *t, AranyaRole **roles,
+                                size_t *roles_len) {
+    AranyaError err;
+
+    *roles_len = BUF_LEN;
+    *roles     = malloc(*roles_len * sizeof(AranyaRole));
+    err        = aranya_query_roles_on_team(&t->clients.operator.client, &t->id,
+                                            *roles, roles_len);
+    EXPECT("error querying roles on team", err);
+    if (roles == NULL) {
+        return ARANYA_ERROR_BUG;
+    }
+
+    return err;
+}
+
+// Query device roles. Returned `roles` ptr must be freed.
+AranyaError query_device_roles(Team *t, AranyaDeviceId *device,
+                               AranyaRole **roles, size_t *roles_len) {
+    AranyaError err;
+
+    *roles_len = BUF_LEN;
+    *roles     = malloc(*roles_len * sizeof(AranyaDeviceId));
```

Comment:
```suggestion
    *roles     = malloc(*roles_len * sizeof(AranyaRole));
```

or use the expression version to help avoid this:

```suggestion
    *roles     = malloc(*roles_len * sizeof(**roles));
```